cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -398,7 +398,7 @@ public String getLeaderInstance(NotificationContext context) {
     public Map<String, String> getLiveHostAndRole(NotificationContext context, String dbName) {
       HelixAdmin admin = context.getManager().getClusterManagmentTool();
       ExternalView view = Utils.getHelixExternalViewWithFixedRetry(admin, cluster, resourceName);
-      Map<String, String> stateMap = view.getStateMap(partitionName);
+      Map<String, String> stateMap = Utils.getStateMap(view, partitionName);
```

Comment:
This is within `getLiveHostAndRole`, and is the one where I observed the NPE in production during failure testing, which motivated this PR. 

Here we are trying to find out the "live" instances in the stateMap for a shard. Therefore, if there is no instances in the stateMap, we should return empty result, instead of crashing and ERROR out. This is a straighforward change imo